core/templates/dev/head/editor/EditorServices.js

```diff
@@ -1495,6 +1493,7 @@ oppia.factory('explorationWarningsService', [
       explorationObjectiveService, INTERACTION_SPECS, WARNING_TYPES,
       FUZZY_RULE_TYPE) {
   var _warningsList = [];
+  var _perStateErrors = [];
```

Comment:
I think the data structure here should be an object whose keys are state names. Each of the corresponding values should be a list of errors. The errors don't need to include the state name.

It may also be a good idea to define a constant for the error strings:

```
oppia.constant('STATE_ERROR_MESSAGES', {
  ADD_INTERACTION: 'Please add an interaction to this card.',
  ...
});
```

and then `_perStateErrors` can take the format:

```
{
  'state1': [STATE_ERROR_MESSAGES.ADD_INTERACTION, ...],
  ...
}
```